src/Enzo/utils/utils.cpp

```diff
@@ -0,0 +1,53 @@
+// See LICENSE_CELLO file for license and copyright information
+
+/// @file     utils.cpp
+/// @author   Matthew Abruzzo (matthewabruzzo@gmail.com)
+/// @date     2024-02-20
+/// @brief    Definitions of generic utility functions useful throughout the
+///           Enzo layer
+
+#include "Enzo/enzo.hpp"
+#include "Enzo/utils/utils.hpp"
+
+#include <cmath>
+
+//----------------------------------------------------------------------
+
+static enzo_float median(enzo_float x, enzo_float y, enzo_float z) noexcept
+{
+  // we intentionally use bitwise-and rather than logical-and for speed
+  if ((x >= y) & (x <= z)) {
+    return x;
+  } else if ((x >= z) & (x <= y)) {
+    return x;
+  } else if ((y >= x) & (y <= z)) {
+    return y;
+  } else if ((y >= z) & (y <= x)) {
+    return y;
+  }
+  return z;
+}
+
+bool enzo_utils::consistent_cube_cellwidths(enzo_float dx, enzo_float dy,
+                                            enzo_float dz) noexcept
+{
+  // we want to ensure the median value is within N ULPs of the other 2 values
+  // - a ulp is the difference between 2 adjacent floating point values
+  //
+  // We'll start with N = 4. But we may want need to change that
```

Comment:
Instead of "We'll start with N = 4. But we may want need to change that" I suggest you say:

We currently define approximate equality to be within 4 ULPs of the median of the three values - this can be seen in the for loop in the helper function.  This may need to be changed if this tolerance ends up being too restrictive.